### PR TITLE
flatten isHelmManaged hook return value 

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:

- flatten `isHelmManaged` 

#### Which issue(s) this PR fixes:

https://app.shortcut.com/replicated/story/68569/refactor-version-history-api-calls-to-use-hooks

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Does this PR require documentation?

none

### CODE REVIEW

File: web/src/components/hooks/useIsHelmManaged.js [hunk 1]
```diff
@@ -1,39 +0,0 @@
-import { useQuery } from "react-query";
-import { Utilities } from "../../utilities/utilities";
-
-async function fetchIsHelmManaged({
-  accessToken = Utilities.getToken(),
-  apiEndpoint = process.env.API_ENDPOINT,
-  _fetch = fetch,
```
Comment (rianfowler) [inline web/src/components/hooks/useIsHelmManaged.js:7]: not doing this anymore. originally i thought i would unit test this stuff but I think just leaving out any logic is the better path.

File: web/src/components/hooks/useIsHelmManaged.js [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-    const res = await _fetch(`${apiEndpoint}/is-helm-managed`, {
-      headers: {
-        Authorization: accessToken,
-        "Content-Type": "application/json",
-      },
-      method: "GET",
-    });
-    if (res.ok) {
-      const response = await res.json();
-      return { isHelmManaged: response.isHelmManaged };
```
Comment (rianfowler) [inline web/src/components/hooks/useIsHelmManaged.js:19]: decided not to do this kind of transformation in the fetch handler. this is technically a mutation that we should unit test and we can do it in the `select`

File: web/src/components/hooks/useIsHelmManaged.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+function UseIsHelmManaged({
+  children,
+}: {
+  children: (props: UseQueryResult) => JSX.Element;
+}) {
+  const query = useIsHelmManaged();
+
+  return children(query);
+}
```
Comment (rianfowler) [inline web/src/components/hooks/useIsHelmManaged.tsx:50]: will be glad when i can get rid of this

File: web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts [hunk 4]
```diff
@@ -47,8 +47,7 @@ export const useSelectedAppClusterDashboard = ({
 }: {
   refetchInterval: number | false;
 }) => {
-  const { data: isHelmManagedResponse } = useIsHelmManaged();
-  const { isHelmManaged } = isHelmManagedResponse || { isHelmManaged: false };
+  const { data: isHelmManaged = false } = useIsHelmManaged();
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts:50]: one thing that kind of bugs me about this is we sometimes have `use-` hooks that you have to pull info off of `{ data: ... }` and some like `useSelectedApp` that just return the data and nothing about the async stuff. I think that inconsistency is bad but not sure how to handle it yet. Maybe we should just pass the `useQuery` interface into `useSelectedApp` but it's composing data from two endpoints.

**Review by miaawong — APPROVED** [review]

**Pull request merged**